database/dataGeneration/stockInfo/stockGenerator.js

```diff
@@ -129,18 +129,17 @@ const generateTags = (number) => {
   for (let i = 0; i < number; i++) {
     returnArr.push(faker.fake("{{commerce.department}}"));
   }
-  return returnArr;
+  return `{${returnArr.join(',')}}`;
 }
 
-const generateStocks = (startIndex, endIndex, cb, finishCB) => {
+const generateStocks = (startIndex, endIndex, cb) => {
   let data = '';
   for (let i = startIndex; i < endIndex; i += 1) {
     let stockCompany;
     i < 100 ? (stockCompany = companyData[i].company) : (stockCompany = faker.company.companyName());
-    data += `${i}, ${tickers[i]},${stockCompany},${generateTags(faker.random.number({ min: 0, max: 3 }))},${faker.random.number({ min: 10000, max: 99999999 })},${faker.random.number({ min: 10000, max: 99999999 })}, \n`;
+    data += `${i}|${tickers[i]}|${faker.finance.amount(200, 400, 2)}|${faker.finance.amount(1, 4, 2)}|"${stockCompany}"|${faker.random.number({ min: 10000, max: 99999999 })}|${faker.finance.amount(0, 1, 2)}|${generateTags(faker.random.number({ min: 0, max: 3 }))}\n`;
```

Comment:
You might consider writing your own random number generator, as the Faker library is [not very efficient](https://kevin.burke.dev/kevin/faker-js-problems/). It may make a difference in your generation time, given the large dataset. However, you may need to go further than Math.random, as there are [several issues there](https://medium.com/@betable/tifu-by-using-math-random-f1c308c4fd9d) as well!

Reply:
@vh02 -- I appreciate the perspective. I read both of the articles you sent, and they seemed to be saying that there may be a uniqueness issue and/or the entire library took 300ms to load (insignificant in the scheme of how long my entire program takes to run), but I didn't see anything on the inefficiency of a single function call. What was your take on the articles?